server/src/test/java/org/elasticsearch/action/ingest/GetPipelineResponseTests.java

```diff
@@ -89,4 +92,41 @@ public void testXContentDeserialization() throws IOException {
             assertEquals(pipelinesMap.get(pipeline.getId()).getConfigAsMap(), pipeline.getConfigAsMap());
         }
     }
+
+    @Override
+    protected GetPipelineResponse doParseInstance(XContentParser parser) throws IOException {
+        return GetPipelineResponse.fromXContent(parser);
+    }
+
+    @Override
+    protected GetPipelineResponse createBlankInstance() {
+        return new GetPipelineResponse();
+    }
+
+    @Override
+    protected GetPipelineResponse createTestInstance() {
+        try {
+            return new GetPipelineResponse(new ArrayList<>(createPipelineConfigMap().values()));
+        } catch (IOException e) {
+            // If we fail to construct an instance we return `null` which would make the user of this method
+            // fail the test.
+            return null;
+        }
+    }
+
+    @Override
+    protected boolean supportsUnknownFields() {
+        return false;
+    }
+
+    @Override
+    protected GetPipelineResponse mutateInstance(GetPipelineResponse response) {
+        try {
+            ArrayList<PipelineConfiguration> clonePipelines = new ArrayList<>(response.pipelines());
+            clonePipelines.add(createRandomPipeline("pipeline_" + clonePipelines.size() + 1));
+            return new GetPipelineResponse(clonePipelines);
+        } catch (IOException e) {
+            return null;
```

Comment:
rather throw UncheckedIOException ?

Reply:
Thanks for the tip. I didn't know these existed.